app/controllers/api/migrations.php

```diff
@@ -0,0 +1,554 @@
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic($migration, Response::MODEL_MIGRATION);
+    });
+
+App::post('/v1/migrations/appwrite/report')
+    ->groups(['api', 'migrations'])
+    ->desc('Generate a report on Appwrite Data')
+    ->label('scope', 'migrations.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'generateAppwriteReport')
+    ->label('sdk.description', '/docs/references/migrations/migration-appwrite-report.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MIGRATION_REPORT)
+    ->param('resources', [], new ArrayList(new WhiteList(Appwrite::getSupportedResources())), 'List of resources to migrate')
+    ->param('endpoint', '', new URL(), "Source's Appwrite Endpoint")
+    ->param('projectID', '', new Text(512), "Source's Project ID")
+    ->param('key', '', new Text(512), "Source's API Key")
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->action(function (array $resources, string $endpoint, string $projectID, string $key, Response $response) {
+
+        $appwrite = new Appwrite($projectID, $endpoint, $key);
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_CREATED)
+            ->dynamic(new Document($appwrite->report($resources)), Response::MODEL_MIGRATION_REPORT);
+    });
+
+
+App::post('/v1/migrations/firebase')
+    ->groups(['api', 'migrations'])
+    ->desc('Migrate Firebase Data')
+    ->label('scope', 'migrations.write')
+    ->label('event', 'migrations.create')
+    ->label('audits.event', 'migration.create')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'migrations')
+    ->label('sdk.method', 'migrateFirebase')
+    ->label('sdk.description', '/docs/references/migrations/migration-firebase.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_MIGRATION)
+    ->param('resources', [], new ArrayList(new WhiteList(Firebase::getSupportedResources())), 'List of resources to migrate')
+    ->param('serviceAccount', '', new Text(512), "Source's Service Account")
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('user')
+    ->inject('events')
+    ->action(function (array $resources, string $serviceAccount, Response $response, Database $dbForProject, Document $project, Document $user, Event $eventsInstance) {
+        $migration = $dbForProject->createDocument('migrations', new Document([
+            '$id' => ID::unique(),
+            'status' => 'pending',
+            'stage' => 'init',
+            'source' => json_encode([
+                'type' => 'firebase',
```

Comment:
Why dont we have constants in the migration library to represent these strings?

Reply:
Sorry, the comment has moved from where it was originally. Could you please correct me on what part of the code this is referrencing?